core/src/main/java/org/wildfly/extras/creaper/core/online/OnlineOptions.java

```diff
@@ -356,6 +356,12 @@ ModelControllerClient createModelControllerClient() throws IOException {
         if (wrappedModelControllerClient != null) {
             return wrappedModelControllerClient;
         }
+        if (protocol == ManagementProtocol.HTTP) {
+            if (username == null || password == null) {
+                throw new IllegalArgumentException("Both username and password must be specified if ManagementProtocol.HTTP is used.");
```

Comment:
Actually, we shouldn't force this. If you remove the `security-realm` attribute from

```xml
            <http-interface security-realm="ManagementRealm" http-upgrade-enabled="true">
                <socket-binding http="management-http"/>
            </http-interface>
```

then the HTTP management interface is available without auth. It's not common, but it's possible. The exception should be thrown only when the first request is made and we find out that auth is required.